vscode_extension/src/commands/copySymbolToClipboard.ts

```diff
@@ -46,13 +52,42 @@ export async function copySymbolToClipboard(
     },
     position,
   };
-  const response = await client.sendRequest<SymbolInformation>(
-    "sorbet/showSymbol",
-    params,
-  );
-
-  await env.clipboard.writeText(response.name);
-  context.log.debug(
-    `Copied symbol name to the clipboard. Name:${response.name}`,
-  );
+
+  let response: SymbolInformation | undefined;
+  if (context.statusProvider.operations.length) {
```

Comment:
What do we check with this condition? Do we check if there are any other requests in flight?

Reply:
> Ideally, the command would only allow one pending operation but that introduced more complexity on the PR (and need to figure out quirks about UX), so decided to do that as follow-up work (note that today, you can queue up multiple requests unknowingly, so this is just making those requests visible).

Reply:
The showSymbol request should be able to complete instantly if the operation is "Typechecking in background..."

Until now, I'm pretty sure that the "Typechecking..." / "Indexing..." / etc. states have all been UI-only. I don't know that the correctness of any features depended on them until this. That suggests to me that we might be solving this at the wrong level of abstraction. That's why I was suggesting possibly implementing this inside Sorbet's LSP mode. That would probably be cleaner, but I think it would require a somewhat widely scoped refactor of some parts of Sorbet's pipeline, it'll have to wait for some other day.

In the mean time, if you hard-code Type checking in background here as not needing a progress bar, that should be fine. Or honestly, you could just show the progress bar even if it was typechecking in background, because it would just go away after a short moment.

Reply:
I'll keep tweaking the experience since even this does not solve all UX issues. As @ilyailya noted, you can queue up multiple operations, which has always been true, but now you might get some UI showing that (making more evident but also noisier). I will look at limiting this when `Typechecking in background...`, as you suggested, in a latter PR.